docs/3_NEW_TEMPLATE.md

```diff
@@ -1,15 +1,406 @@
+
+| New CIT Name | Description | Deployment Goal |
+|----------|----------|----------|
+| **az-hello-world-from-scratch** | A Cobalt Infrastructue Template that when ran creates a basic [Azure Function App](https://docs.microsoft.com/en-us/azure/azure-functions/functions-overview) within an [App Service Plan](https://docs.microsoft.com/en-us/azure/app-service/overview-hosting-plans) accompanied with [Azure Storage](https://azure.microsoft.com/en-us/services/storage/blobs/). | <image src="https://user-images.githubusercontent.com/10041279/67136958-a9d27600-f1f3-11e9-896c-d18f3a287de5.png" width="500" height="200"/> |
+
+### **Step 2:** Plan your modular strategy
+
+Once you've modeled your planned infrastructure resources, we recommend answering the following questions:
+
+1. > QUESTION: **"Which portion of my planned infrastructure do I want to roll-up into a reusable module?"**
+
+    **ANSWER:** Currently, most Cobalt Modules are scoped to a very specific cloud infrastructure resource. In other words, modules encapsulate a configuration that targets a providers very specific cloud service product. The Azure Function Module we are creating will follow this pattern.
+
+1. > QUESTION: **"Does Cobalt have existing reusuable modules configured for any portion of my planned infrastructure?"**
+
+    **ANSWER:** At the time of this walkthrough, there are reusable modules for an Azure App Service Plan and Azure Storage, so you will use those to build part of your *CIT*. However, there is no current *Cobalt Module* configured for deploying an Azure Function. So, yes, a new Cobalt Module needs to be created. Let's design one!
+
+### **Step 3:** Design Your Terraform Based *Cobalt Module*s
+
+The first step of designing a *Cobalt Module* involves defining a Terraform module's 3 primary elements: input variables, output variables and resources. This will be all done via Terraform's [HCL language](https://learn.hashicorp.com/terraform), a language that grants developers the ability to target multiple cloud providers. Documentation for the HCL language is partitioned by cloud provider. You will become very familiar with Terraform's cloud provider documentation as you learn to use and build your own modules and CITs.
+
+1. **Visit the below link**. Use the documentation at the link as a reference for insight into how we are making the below design decisions for the Azure Function *Cobalt Module* in this walkthrough.
+
+    * Terraform - [Azure ARM Provider - Azure Function](https://www.terraform.io/docs/providers/azurerm/r/function_app.html#example-usage-in-a-consumption-plan-)
+
+1. **Define your resources** - Defined below are the resource blocks that will be implemented:
+
+    | Resource | Description |
```

Comment:
This may be nice with another column that describes which module each will live in. it makes the description less wordy, and goes along with the "module planning" strategy you are discussing here

Reply:
Definitely added a "module" column. Good thinking.